### PR TITLE
Namespace documentation for Kestrel namespaces

### PR DESCRIPTION
**Pull request by sebastienros**
Addresses dotnet/aspnetcore#27286

### CODE REVIEW

**Comment by sebastienros** [comment]
I am not sure if I need to remove the links that point to the namespace class lists since I can now see in the preview that these remarks appear in this page too. But if these docs appear somewhere else then that could still be nice to be able to go to the namespace class lists directly.

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Server.Kestrel.Core.Features">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provides types that are used to describe feature components in Kestrel web server.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml:3]:
```suggestion
    <summary>Provides types that are used to describe feature components in the Kestrel web server.</summary>
```

**Review by Rick-Anderson — APPROVED** [review]

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml (hunk 1 above)
Comment (sebastienros) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml:3]: Will apply this suggestion in other files

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Server.Kestrel.Core.Features">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provides types that are used to describe feature components in the Kestrel web server.</summary>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.server.kestrel.core.features">Microsoft.AspNetCore.Server.Kestrel.Core.Features Namespace</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml:4]: I believe this is just linking to the page you're already on. I think you should either remove these namespace links, or change them to link to https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel.

**Review by JamesNK — CHANGES_REQUESTED** [review]

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml (hunk 2 above)
Comment (Rick-Anderson) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml:4]: ~/en-us~
https://docs.microsoft.com/aspnet/core/fundamentals/servers/kestrel

Comment (sebastienros) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml]: Updated everywhere

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.xml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Server.Kestrel.Core">
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Provides types that are used to customize the behavior of the Kestrel web server.</summary>
     <remarks>To be added.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.xml:4]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/fundamentals/servers/kestrel#kestrel-options">Kestrel options</see>.</remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Server.Kestrel.Core">
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Provides types that are used to customize the behavior of the Kestrel web server.</summary>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.xml:3]:
```suggestion
    <summary>Provides options types such as <see cref="T:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions" /> and <see cref="T:Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions" /> that are used to customize the behavior of the Kestrel web server.</summary>
```

**Review by JamesNK — APPROVED** [review]

**Comment by sebastienros** [comment]
@JamesNK did I forget to react to some comments?

**Comment by JamesNK** [comment]
No, just some extra changes I added just now.

**Comment by sebastienros** [comment]
Ah, you found actual articles for these, thanks

**Comment by scottaddie** [comment]
Docs Build status updates of commit _[4ce325b](https://github.com/dotnet/AspNetApiDocs/commits/4ce325b6df4f032a33b259fb1b424dccd381aac4)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.Features.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core.features?view=aspnetcore-2.0&branch=pr-en-us-176) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Core.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core?view=aspnetcore-2.0&branch=pr-en-us-176) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Https.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Https.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.https?view=aspnetcore-1.0&branch=pr-en-us-176) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.transport.experimental.quic?view=aspnetcore-5.0&branch=pr-en-us-176) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.transport.libuv?view=aspnetcore-2.0&branch=pr-en-us-176) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.transport.sockets?view=aspnetcore-2.1&branch=pr-en-us-176) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.xml](https://github.com/dotnet/AspNetApiDocs/blob/sebros/ns-docs/aspnet-core/xml/ns-Microsoft.AspNetCore.Server.Kestrel.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel?view=aspnetcore-1.0&branch=pr-en-us-176) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C4%5Cbaf25683-9e81-c379-8ba7-62f30178bab2%5CPullRequest%5C202012042007261767-176%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=%2BIn1O6QdtcglUNInz5PLocx%2FGT0Ke4xq49nCS7gFzPc%3D&st=2020-12-04T20%3A05%3A39Z&se=2021-01-04T20%3A10%3A39Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/baf25683-9e81-c379-8ba7-62f30178bab2?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Pull request merged**